aws/aws.go

```diff
@@ -29,6 +33,96 @@ const subscriptionIDMaxLength = 40
 // convention we're limited to the total 80-character max of the SQS name length.
 const topicNameMaxLength = 40 - len(topicNamePrefix)
 
+// VerifyPermissions checks if the aws config exists and checks if it has permissions to
+// create sns topics, send messages, create SQS topics, delete topics, and delete sqs queues
+func VerifyPermissions(cfg *aws.Config) error {
+
+	// Check if environment contains aws config
+	_, errConfig := cfg.Credentials.Get()
+	if errConfig != nil {
+		return errConfig
+	}
+
+	topic := fmt.Sprintf("verifyPermissions")
+	subscriptionID := "verifySubsId"
```

Comment:
this needs to be globally unique: a race condition can exist if this function is called by different processes at the same time. Add a UUID to it.

Reply:
Additionally, we should give the topic a unique name as well, or add another UUID to be part of the message metadata and to filter the subscription on it. 

Otherwise, with a shared topic like `verifyPermissions`, we could inadvertently receive a verify message from a different invocation.